subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -561,10 +605,7 @@ static int scene_srv_init(struct bt_mesh_model *model)
 {
 	struct bt_mesh_scene_srv *srv = model->user_data;
 
-	if (model->id == BT_MESH_MODEL_ID_SCENE_SETUP_SRV) {
-		srv->setup_mod = model;
-		return 0;
-	}
+	sys_slist_prepend(&scene_servers, &srv->n);
```

Comment:
`scene_servers` needs to be reset in `scene_srv_reset`.

Reply:
I don't think so, the init call will not be called again after a reset, it just calls start